pkg/exthttp/hedging.go

```diff
@@ -0,0 +1,73 @@
+package exthttp
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/caio/go-tdigest"
+	"github.com/cristalhq/hedgedhttp"
+)
+
+type CustomBucketConfig struct {
+	HedgingConfig HedgingConfig `yaml:"hedging_config"`
+}
+
+type HedgingConfig struct {
+	UpTo uint `yaml:"up_to"`
+}
+
+var HedgedOptions CustomBucketConfig
+
+type durationRoundTripper struct {
+	Transport http.RoundTripper
+	TDigest   *tdigest.TDigest
+	mu        sync.Mutex
+}
+
+func (drt *durationRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, err := drt.Transport.RoundTrip(req)
+	if err != nil {
+		return nil, err
+	}
+	duration := time.Since(start).Seconds()
+	drt.mu.Lock()
+	err = drt.TDigest.Add(duration)
+	if err != nil {
+		return nil, err
+	}
+	drt.mu.Unlock()
+	return resp, err
+}
+
+func (drt *durationRoundTripper) nextFn() (int, time.Duration) {
+	drt.mu.Lock()
+	defer drt.mu.Unlock()
+
+	delaySec := drt.TDigest.Quantile(0.9)
```

Comment:
I wonder if we should make this percentile configurable as well?

Reply:
Yes, we can make it configurable and default to 90th percentile